tests_tf/test_attacks.py

```diff
@@ -250,6 +268,69 @@ def test_generate_np_clip_works_as_expected(self):
     self.assertClose(np.min(x_adv), -0.2)
     self.assertClose(np.max(x_adv), 0.1)
 
+class TestFastGradientMethod(CommonAttackProperties):
+
+  def setUp(self):
+    super(TestFastGradientMethod, self).setUp()
+
+    self.attack = FastGradientMethod(self.model, sess=self.sess)
+
+  def test_optimize_linear_l1(self):
+
+    # This test makes sure that `optimize_linear` actually finds the optimal
+    # perturbation for ord=1.
+    # A common misconcpetion is that FGM for ord=1 consists of dividing
+    # the gradient by its L1 norm.
+    # If you do that for the problem in this unit test, you'll get an
+    # objective function value of ~1.667. The optimal result is 2.
+
+    # We need just one example in the batch and two features to show the
+    # common misconception is suboptimal.
+    grad = tf.placeholder(tf.float32, [1, 2])
```

Comment:
Here and in the next test, maybe add `shape=`:
```
grad = tf.placeholder(tf.float32, shape=[1, 2])
```

Otherwise it might be confusing to read this line if you don't remember exact API contract for tf.placeholder.